src/components/TimeseriesChartExport/TimeseriesChartExport.tsx

```diff
@@ -0,0 +1,417 @@
+  onSuccess?: () => void;
+  formItemLayout?: FormItemLayout;
+  theme?: AnyIfEmpty<{}>;
+  strings?: PureObject;
+}
+
+const defaultStrings: PureObject = {
+  title: 'Export chart data',
+  labelRange: 'Range',
+  labelGranularity: 'Label Granularity',
+  labelGranularityHelp: 'Example inputs: 15s, 1m, 5h, 2d',
+  formatTimestamp: 'Format timestamp?',
+  formatTimestampHelp: 'e.g. 2018-04-02 12:20:20',
+  delimiterLabel: 'Select delimiter',
+  delimiterHelp: 'The character that will separate your data fields',
+  csvDownload: 'Download as CSV',
+  csvDownloadInProgress: 'Download as CSV',
+  closeBtn: 'Close',
+  imageDownloadLabel: 'Image download',
+  imageDownloadBtn: 'Download as SVG',
+  cellLimitErr:
+    'You hit the limit of {{ cellLimit }} datapoints - some data may be omitted',
+};
+
+// TODO: Check tree shacking for TimeseriesChartExport component
+const { RangePicker } = DatePicker;
+const CELL_LIMIT = 10000;
+const formatData = 'YYYY-MM-DD HH:mm:ss';
+const formItemLayoutDefault: FormItemLayout = {
+  labelCol: { xs: { span: 24 }, sm: { span: 8 } },
+  wrapperCol: { xs: { span: 24 }, sm: { span: 16 } },
+};
+
+const isGreaterThenLimit = (
+  limit: number,
+  start: number,
+  end: number,
+  granularity: number,
+  seriesNumber: number
+): boolean => {
+  if (!start || !end || !granularity) {
+    return false;
+  }
+
+  const requestedNumberOfPoints = ((end - start) / granularity) * seriesNumber;
+
+  return requestedNumberOfPoints > limit;
+};
+
+// tslint:disable-next-line:no-big-function
+const TimeseriesChartExport = (props: TimeseriesChartExportProps) => {
+  const {
+    form,
+    form: { getFieldDecorator },
+    formItemLayout = formItemLayoutDefault,
+    timeseriesIds,
+    visible,
+    defaultRange,
+    granularity,
+    modalWidth = 600,
+    cellLimit = 10000,
+    downloadAsSvg,
+    hideModal,
+    fetchCSV,
+    retrieveTimeseries,
+    onError,
+    onSuccess,
+    strings = {},
+  } = props;
+  const context = useContext(ClientSDKContext);
+  const [limit, setLimit] = useState(cellLimit);
+  const [limitHit, setLimitHit] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [series, setSeries]: [TimeSeries[], any] = useState([]);
+  const lang = { ...defaultStrings, ...strings };
+  const {
+    title,
+    labelRange,
+    cellLimitErr,
+    labelGranularity,
+    labelGranularityHelp,
+    formatTimestamp,
+    formatTimestampHelp,
+    delimiterLabel,
+    delimiterHelp,
+    csvDownload,
+    csvDownloadInProgress,
+    closeBtn,
+    imageDownloadLabel,
+    imageDownloadBtn,
+  } = lang;
+
+  const fetchCSVCall: FetchCSVCall = async (
+    request,
+    { aggregate, delimiter, readableDate, granularity: granularityString }
+  ) => {
+    const data = await context!.datapoints.retrieve(request);
+    const format = readableDate ? formatData : '';
+
+    return datapointsToCSV({
+      data,
+      aggregate,
+      delimiter,
+      format,
+      granularity: granularityString,
```

Comment:
i guess you renamed a property here but vs code didn't fix it the way you wanted

Reply:
should be just `granularity`

Reply:
no, cause we have `granularity` name from props, and here granularity like shadow variable